pkg/utils/apiutil/serverapi/middleware.go

```diff
@@ -200,11 +202,8 @@ func (h *redirector) ServeHTTP(w http.ResponseWriter, r *http.Request, next http
 			return
 		}
 		clientUrls = append(clientUrls, targetAddr)
-		// Add a header to the response, this is not a failure injection
-		// it is used for testing, to check whether the request is forwarded to the micro service
-		failpoint.Inject("checkHeader", func() {
-			w.Header().Set(apiutil.ForwardToMicroServiceHeader, "true")
-		})
+		// Add a header to the response, it is used to mark whether the request has been forwarded to the micro service.
+		w.Header().Add(apiutil.XForwardedToMicroServiceHeader, "true")
```

Comment:
Can we not put this header on every forward request? This reduces network traffic. For example, when the `len(r.Header.Get(apiutil.XForbiddenForwardToMicroServiceHeader)) > 0`, that's the same as making an extra mark, similar to failpoint.